app/src/androidTest/java/com/example/samplegallery/PhotoBlowupFragmentTest.java

```diff
@@ -0,0 +1,140 @@
+package com.example.samplegallery;
+
+import android.support.test.InstrumentationRegistry;
+import android.support.test.espresso.Espresso;
+import android.support.test.espresso.IdlingResource;
+import android.support.test.filters.LargeTest;
+import android.support.test.rule.ActivityTestRule;
+import android.support.test.runner.AndroidJUnit4;
+import android.util.Log;
+
+import com.android.volley.Request;
+import com.android.volley.RequestQueue;
+import com.example.samplegallery.Utilities.VolleyRequestQueue;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.lang.reflect.Field;
+import java.util.Set;
+
+import static android.support.test.espresso.Espresso.onData;
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.assertion.ViewAssertions.matches;
+import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+import static android.support.test.espresso.matcher.ViewMatchers.withResourceName;
+import static org.hamcrest.CoreMatchers.allOf;
+import static org.hamcrest.CoreMatchers.anything;
+import static org.hamcrest.CoreMatchers.not;
+
+/**
+ * Created by Sean Walker on 1/29/17.
+ */
+
+
+// test photo blowup synchronization
+@RunWith(AndroidJUnit4.class)
+@LargeTest
+public class PhotoBlowupFragmentTest {
+    // disable the volley queue before the activity is launched to make sure that we may test
+    // the behaviour of the test both before and after the data has been pulled.
+    @Rule
+    public ActivityTestRule<MainActivity> mActivityRule =
+            new ActivityTestRule<MainActivity>(MainActivity.class) {
+
+                @Override
+                protected void beforeActivityLaunched() {
+                    super.beforeActivityLaunched();
+
+                    VolleyRequestQueue
+                            .getInstance(InstrumentationRegistry.getTargetContext().getApplicationContext())
+                            .getRequestQueue()
+                            .stop();
+                }
+            };
+    @Test
+    public void photoBlowupTest() throws Exception {
+        // register our volley request queue as an idling resource and start it.
+        VolleyQueueIdlingResource vqidr = new VolleyQueueIdlingResource("Volley Queue");
+        VolleyRequestQueue.getInstance(null).getRequestQueue().start();
```

Comment:
What about the time BEFORE you start pulling images? You probably want your spinner to be visible too (you should test for that as well!).